diffoscope itself is still failing to build in its tests:

```
diffoscope> ==================================== ERRORS ====================================
diffoscope> ________________________ ERROR at setup of test_zipinfo ________________________
diffoscope>
diffoscope> apk1 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test1.apk>
diffoscope> apk2 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test2.apk>
diffoscope>
diffoscope>     @pytest.fixture
diffoscope>     def differences(apk1, apk2):
diffoscope> >       return apk1.compare(apk2).details
diffoscope>
diffoscope> tests/comparators/test_apk.py:54:
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope> diffoscope/comparators/zip.py:234: in compare
diffoscope>     x = super().compare(other, source)
diffoscope> diffoscope/comparators/utils/file.py:536: in compare
diffoscope>     difference = self._compare_using_details(other, source)
diffoscope> diffoscope/comparators/utils/file.py:437: in _compare_using_details
diffoscope>     details.extend(self.compare_details(other, source))
diffoscope> diffoscope/comparators/apk.py:274: in compare_details
diffoscope>     get_v2_signing_keys(self.path),
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope>
diffoscope> path = '/build/diffoscope-288/tests/data/test1.apk'
diffoscope>
diffoscope>     def get_v2_signing_keys(path):
diffoscope> >       from androguard.core.bytecodes import apk
diffoscope> E       ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope>
diffoscope> diffoscope/comparators/apk.py:323: ModuleNotFoundError
diffoscope> __________________ ERROR at setup of test_apk_metadata_source __________________
diffoscope>
diffoscope> apk1 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test1.apk>
diffoscope> apk2 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test2.apk>
diffoscope>
diffoscope>     @pytest.fixture
diffoscope>     def differences(apk1, apk2):
diffoscope> >       return apk1.compare(apk2).details
diffoscope>
diffoscope> tests/comparators/test_apk.py:54:
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope> diffoscope/comparators/zip.py:234: in compare
diffoscope>     x = super().compare(other, source)
diffoscope> diffoscope/comparators/utils/file.py:536: in compare
diffoscope>     difference = self._compare_using_details(other, source)
diffoscope> diffoscope/comparators/utils/file.py:437: in _compare_using_details
diffoscope>     details.extend(self.compare_details(other, source))
diffoscope> diffoscope/comparators/apk.py:274: in compare_details
diffoscope>     get_v2_signing_keys(self.path),
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope>
diffoscope> path = '/build/diffoscope-288/tests/data/test1.apk'
diffoscope>
diffoscope>     def get_v2_signing_keys(path):
diffoscope> >       from androguard.core.bytecodes import apk
diffoscope> E       ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope>
diffoscope> diffoscope/comparators/apk.py:323: ModuleNotFoundError
diffoscope> ____________ ERROR at setup of test_skip_undecoded_android_manifest ____________
diffoscope>
diffoscope> apk1 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test1.apk>
diffoscope> apk2 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test2.apk>
diffoscope>
diffoscope>     @pytest.fixture
diffoscope>     def differences(apk1, apk2):
diffoscope> >       return apk1.compare(apk2).details
diffoscope>
diffoscope> tests/comparators/test_apk.py:54:
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope> diffoscope/comparators/zip.py:234: in compare
diffoscope>     x = super().compare(other, source)
diffoscope> diffoscope/comparators/utils/file.py:536: in compare
diffoscope>     difference = self._compare_using_details(other, source)
diffoscope> diffoscope/comparators/utils/file.py:437: in _compare_using_details
diffoscope>     details.extend(self.compare_details(other, source))
diffoscope> diffoscope/comparators/apk.py:274: in compare_details
diffoscope>     get_v2_signing_keys(self.path),
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope>
diffoscope> path = '/build/diffoscope-288/tests/data/test1.apk'
diffoscope>
diffoscope>     def get_v2_signing_keys(path):
diffoscope> >       from androguard.core.bytecodes import apk
diffoscope> E       ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope>
diffoscope> diffoscope/comparators/apk.py:323: ModuleNotFoundError
diffoscope> __________________ ERROR at setup of test_no_android_manifest __________________
diffoscope>
diffoscope> apk1 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test1.apk>
diffoscope> apk3 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test3.apk>
diffoscope>
diffoscope>     @pytest.fixture
diffoscope>     def differences2(apk1, apk3):
diffoscope> >       return apk1.compare(apk3).details
diffoscope>
diffoscope> tests/comparators/test_apk.py:59:
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope> diffoscope/comparators/zip.py:234: in compare
diffoscope>     x = super().compare(other, source)
diffoscope> diffoscope/comparators/utils/file.py:536: in compare
diffoscope>     difference = self._compare_using_details(other, source)
diffoscope> diffoscope/comparators/utils/file.py:437: in _compare_using_details
diffoscope>     details.extend(self.compare_details(other, source))
diffoscope> diffoscope/comparators/apk.py:274: in compare_details
diffoscope>     get_v2_signing_keys(self.path),
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope>
diffoscope> path = '/build/diffoscope-288/tests/data/test1.apk'
diffoscope>
diffoscope>     def get_v2_signing_keys(path):
diffoscope> >       from androguard.core.bytecodes import apk
diffoscope> E       ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope>
diffoscope> diffoscope/comparators/apk.py:323: ModuleNotFoundError
diffoscope> =================================== FAILURES ===================================
diffoscope> _____________________________ test_no_differences ______________________________
diffoscope>
diffoscope> apk1 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test1.apk>
diffoscope>
diffoscope>     def test_no_differences(apk1):
diffoscope> >       difference = apk1.compare(apk1)
diffoscope>
diffoscope> tests/comparators/test_apk.py:48:
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope> diffoscope/comparators/zip.py:234: in compare
diffoscope>     x = super().compare(other, source)
diffoscope> diffoscope/comparators/utils/file.py:536: in compare
diffoscope>     difference = self._compare_using_details(other, source)
diffoscope> diffoscope/comparators/utils/file.py:437: in _compare_using_details
diffoscope>     details.extend(self.compare_details(other, source))
diffoscope> diffoscope/comparators/apk.py:274: in compare_details
diffoscope>     get_v2_signing_keys(self.path),
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope>
diffoscope> path = '/build/diffoscope-288/tests/data/test1.apk'
diffoscope>
diffoscope>     def get_v2_signing_keys(path):
diffoscope> >       from androguard.core.bytecodes import apk
diffoscope> E       ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope>
diffoscope> diffoscope/comparators/apk.py:323: ModuleNotFoundError
diffoscope> __________________________ test_compare_non_existing ___________________________
diffoscope>
diffoscope> monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7ffff631afc0>
diffoscope> apk1 = <<class 'abc.ApkFile'> /build/diffoscope-288/tests/data/test1.apk>
diffoscope>
diffoscope>     @skip_unless_tools_exist("apktool", "zipinfo")
diffoscope>     def test_compare_non_existing(monkeypatch, apk1):
diffoscope> >       assert_non_existing(monkeypatch, apk1)
diffoscope>
diffoscope> tests/comparators/test_apk.py:64:
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope> tests/utils/nonexisting.py:31: in assert_non_existing
diffoscope>     difference = fixture.compare(MissingFile("/nonexisting", fixture))
diffoscope> diffoscope/comparators/zip.py:234: in compare
diffoscope>     x = super().compare(other, source)
diffoscope> diffoscope/comparators/utils/file.py:536: in compare
diffoscope>     difference = self._compare_using_details(other, source)
diffoscope> diffoscope/comparators/utils/file.py:437: in _compare_using_details
diffoscope>     details.extend(self.compare_details(other, source))
diffoscope> diffoscope/comparators/apk.py:274: in compare_details
diffoscope>     get_v2_signing_keys(self.path),
diffoscope> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
diffoscope>
diffoscope> path = '/build/diffoscope-288/tests/data/test1.apk'
diffoscope>
diffoscope>     def get_v2_signing_keys(path):
diffoscope> >       from androguard.core.bytecodes import apk
diffoscope> E       ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope>
diffoscope> diffoscope/comparators/apk.py:323: ModuleNotFoundError
diffoscope> =========================== short test summary info ============================
diffoscope> FAILED tests/comparators/test_apk.py::test_no_differences - ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope> FAILED tests/comparators/test_apk.py::test_compare_non_existing - ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope> ERROR tests/comparators/test_apk.py::test_zipinfo - ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope> ERROR tests/comparators/test_apk.py::test_apk_metadata_source - ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope> ERROR tests/comparators/test_apk.py::test_skip_undecoded_android_manifest - ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope> ERROR tests/comparators/test_apk.py::test_no_android_manifest - ModuleNotFoundError: No module named 'androguard.core.bytecodes'
diffoscope> = 2 failed, 683 passed, 39 skipped, 5 deselected, 1 xfailed, 2 xpassed, 4 errors in 81.00s (0:01:21) =
```

--- a/pkgs/tools/misc/diffoscope/default.nix
+++ b/pkgs/tools/misc/diffoscope/default.nix
@@ -106,11 +106,11 @@ in
 # Note: when upgrading this package, please run the list-missing-tools.sh script as described below!
 python.pkgs.buildPythonApplication rec {
   pname = "diffoscope";
-  version = "287";
+  version = "288";
 
   src = fetchurl {
     url = "https://diffoscope.org/archive/diffoscope-${version}.tar.bz2";
-    hash = "sha256-0s7pT8pAMCE+csd9/+Dv4AbCK0qxDacQ9fNcMYCNDbw=";
+    hash = "sha256-HeJfqjmuduVc/C0kfaAsiQuoKrrwlW69lkfYDz1uFlg=";
   };
 
   outputs = [
@@ -121,11 +121,11 @@ python.pkgs.buildPythonApplication rec {
   patches = [ ./ignore_links.patch ];
 
   postPatch = ''
-    # Upstream doesn't provide a PKG-INFO file
-    sed -i setup.py -e "/'rpm-python',/d"
-
     # When generating manpage, use the installed version
-    substituteInPlace doc/Makefile --replace "../bin" "$out/bin"
+    substituteInPlace doc/Makefile --replace-fail "../bin" "$out/bin"
+
+    substituteInPlace diffoscope/comparators/apk.py \
+      --replace-fail "from androguard.core.bytecodes import apk" "from androguard.core import apk"
   '';
 
   nativeBuildInputs = [

